### DISCUSSION TITLE
How to update xmake in package manager after using xmake update?

### DISCUSSION THREAD

**Discussion by f-zl** [xmake-io/xmake#4426] [Q&A] [answered]
I installed xmake in msys2 using pacman. I met the problem https://github.com/xmake-io/xmake/issues/4278 and used `xmake update -s dev` to solve it. After that, pacman cannot update xmake because of conflicting files.

> error: failed to commit transaction (conflicting files)
mingw-w64-x86_64-xmake: /mingw64/share/xmake/actions/test/main.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/actions/test/xmake.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/core/base/graph.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/core/sandbox/modules/import/core/base/graph.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_cflags.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_cfuncs.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_cincludes.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_csnippets.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_ctypes.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_cxxflags.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_cxxfuncs.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_cxxincludes.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_cxxsnippets.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_cxxtypes.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_features.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_links.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_macros.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/check_syslinks.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/check/xmake.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/qt/qt_add_static_plugins.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/includes/qt/xmake.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/modules/private/utils/target.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/platforms/applexros/xmake.lua exists in filesystem
mingw-w64-x86_64-xmake: /mingw64/share/xmake/toolchains/xcode/load_applexros.lua exists in filesystem

I tried reinstalling xmake but it failed in the same way. Is there an easy way to update xmake in pacman after `xmake update -s dev`?

**Comment by waruqi** [comment]
try 

```
xmake update dev
source ~/.xmake/profile
```

**Reply by f-zl** [reply, marked as answer]
Thanks for the reply. Though `xmake update dev` seems to download a GUI installer rather than a msys2 package.

I manage to reinstall xmake by
1. Remove pacman package: `pacman -R mingw-w64-x86_64-xmake`
2. Remove remaining files in /share/xmake: `rm -rf /mingw64/share/xmake`
3. Reinstall:  `pacman -S mingw-w64-x86_64-xmake`
